src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -296,11 +296,14 @@ protected virtual void ValidateCreateChannelParameters(EndpointAddress remoteAdd
 
         protected override TChannel OnCreateChannel(EndpointAddress remoteAddress, Uri via)
         {
-            //EndpointAddress httpRemoteAddress = remoteAddress != null && WebSocketHelper.IsWebSocketUri(remoteAddress.Uri) ?
-            //    new EndpointAddress(WebSocketHelper.NormalizeWsSchemeWithHttpScheme(remoteAddress.Uri), remoteAddress) :
-            //    remoteAddress;
+            if (typeof (TChannel) != typeof (IRequestChannel))
+            {
+                remoteAddress = remoteAddress != null && !WebSocketHelper.IsWebSocketUri(remoteAddress.Uri) ?
+                    new EndpointAddress(WebSocketHelper.NormalizeHttpSchemeWithWsScheme(remoteAddress.Uri), remoteAddress) :
+                    remoteAddress;
+                via = !WebSocketHelper.IsWebSocketUri(via) ? WebSocketHelper.NormalizeHttpSchemeWithWsScheme(via) : via;
+            }
```

Comment:
Does this difference have any affect on existing Windows Store apps (e.g. in the URL's they use)?  Or it is strictly an internal implementation difference (existing) customer apps would never see?